onnxruntime/core/providers/webgpu/nn/layer_norm.h

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/providers/webgpu/program.h"
+#include "core/providers/webgpu/webgpu_kernel.h"
+
+namespace onnxruntime {
+namespace webgpu {
+
+class LayerNormProgram final : public Program<LayerNormProgram> {
+ public:
+  LayerNormProgram(int64_t axis,
+                   float epsilon,
+                   int64_t stash_type,
+                   bool has_bias,
+                   size_t x_size,
+                   bool is_fp16,
+                   bool simplified) : Program{"LayerNorm"},
+                                      axis_{axis},
+                                      epsilon_{epsilon},
+                                      stash_type_{stash_type},
+                                      has_bias_{has_bias},
+                                      x_size_{x_size},
+                                      is_fp16_{is_fp16},
+                                      simplified_{simplified} {}
+
+  Status GenerateShaderCode(ShaderHelper& sh) const override;
+
+  WEBGPU_PROGRAM_DEFINE_UNIFORM_VARIABLES(
+      {"norm_count", ProgramUniformVariableDataType::Uint32},
+      {"norm_size", ProgramUniformVariableDataType::Uint32},
+      {"norm_size_vectorized", ProgramUniformVariableDataType::Uint32},
+      {"epsilon", ProgramUniformVariableDataType::Float32});
+
+ private:
+  int64_t axis_;
+  float epsilon_;
```

Comment:
I was trying to build WebGPU EP on android, and it is raising  -Wunused-private-field error for `axis_`, `epsilon_`, `stash_type_`, `x_size_`

Reply:
thank you for reporting and I will go with a fix. Would you like to help to create a WebGPU EP CI pipeline for Android?